packages/netlify-cms-backend-bitbucket/src/API.ts

```diff
@@ -551,38 +551,57 @@ export default class API {
     const branch = branchFromContentKey(contentKey);
     const pullRequest = await this.getBranchPullRequest(branch);
     const diff = await this.getDifferences(branch);
-    const { newPath: path, newFile } = diff.find(d => !d.binary) as {
-      newPath: string;
-      newFile: boolean;
-    };
+    const entries = diff.filter(d => !d.binary).map(d => ({ path: d.newPath, newFile: d.newFile }));
     // TODO: get real file id
     const mediaFiles = await Promise.all(
-      diff.filter(d => d.newPath !== path).map(d => ({ path: d.newPath, id: null })),
+      diff.filter(d => d.binary).map(d => ({ path: d.newPath, id: null })),
     );
     const label = await this.getPullRequestLabel(pullRequest.id);
     const status = labelToStatus(label);
-    return { branch, collection, slug, path, status, newFile, mediaFiles };
+    return { branch, collection, slug, status, entries, mediaFiles };
   }
 
-  async readUnpublishedBranchFile(contentKey: string) {
-    const {
-      branch,
-      collection,
-      slug,
-      path,
-      status,
-      newFile,
-      mediaFiles,
-    } = await this.retrieveMetadata(contentKey);
+  async readUnpublishedBranchFile(contentKey: string, loadEntryMediaFiles) {
+    const { branch, collection, slug, status, entries, mediaFiles } = await this.retrieveMetadata(
+      contentKey,
+    );
 
-    const fileData = (await this.readFile(path, null, { branch })) as string;
+    if (entries.length === 1) {
```

Comment:
Could we just return an array here? The backend seems to change it into an array anyway here https://github.com/netlify/netlify-cms/pull/3366/files#diff-98047a59f166f83c042f0f4d7589d550R690